django/test/client.py

```diff
@@ -226,7 +226,16 @@ def is_file(thing):
 def encode_file(boundary, key, file):
     def to_bytes(s):
         return force_bytes(s, settings.DEFAULT_CHARSET)
-    filename = os.path.basename(file.name) if hasattr(file, 'name') else ''
```

Comment:
I considered this, but I thought that expanding the ternary made the logic a bit clearer.

I tend to find that once you start adding compound expressions to ternaries things can get a bit dense and difficult to follow.

However, I'm happy to go with your suggestion if this violates the house style, or if anybody has a strong view.